packages/library/src/commonMain/kotlin/io/realm/internal/SuspendableNotifier.kt

```diff
@@ -34,7 +31,7 @@ import kotlinx.coroutines.withContext
  * inside writes. Users can therefor not register change listeners on live objects, but it is assumed that other
  * layers check that invariant before methods on this class are called.
  */
-internal class SuspendableNotifier(private val owner: Realm, private val dispatcher: CoroutineDispatcher) {
+internal class SuspendableNotifier(private val owner: RealmImpl, private val dispatcher: CoroutineDispatcher) {
```

Comment:
Couldn't we just keep the Realm interface here?

Reply:
The internal classes accesses internal stuff. In this case the `owner.configuration.nativeConfig`. This is only exposed because the `RealmImpl` overrides `configuration` offering access to the configuration as a `RealmConfigurationImpl`. 

Alternatively, we need to cast all the public types when passing them to internal classes, but found it easiest to just use the Impl-types as soon as we have them. But technically the internal API should also be separated into an interface and implementation, as we could end up in troubles separating API from implementation at a later stage. 

@edualonso @cmelchior Any better patterns for this?

Reply:
Hmm, yeah, I can see it being tricky. It is probably fine to leave this as it is? Then we can iterate on it in a later stage?